src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,661 @@
+    fprintf(outfile, "RHS fn evals                 = %ld\n", step_mem->nfe);
+    fprintf(outfile, "RHS fn evals for Dom. Eigs.  = %ld\n",
+            step_mem->dom_eig_nfe);
+    fprintf(outfile, "Number of dom_eig updates    = %ld\n",
+            step_mem->num_dom_eig_updates);
+    fprintf(outfile, "Avr. num. of stages taken    = %.2f\n", avg_stage);
+    fprintf(outfile, "Max. num. of stages taken    = %d\n", step_mem->stage_max);
+    fprintf(outfile, "Max. num. of stages allowed  = %d\n",
+            step_mem->stage_max_limit);
+
+    fprintf(outfile, "Max. spectral radius         = %.2f\n", step_mem->spr_max);
+    fprintf(outfile, "Min. spectral radius         = %.2f\n", step_mem->spr_min);
+    break;
+  case SUN_OUTPUTFORMAT_CSV:
+    fprintf(outfile, ",RHS fn evals,%ld", step_mem->nfe);
+    fprintf(outfile, ",RHS fn evals for Dom. Eigs.,%ld", step_mem->dom_eig_nfe);
+    fprintf(outfile, ",Number of dom_eig update calls,%ld",
+            step_mem->num_dom_eig_updates);
+    fprintf(outfile, ",Avr. num. of stages taken,%.2f", avg_stage);
+    fprintf(outfile, ",Max. num. of stages taken,%d", step_mem->stage_max);
+    fprintf(outfile, ",Max. num. of stages allowed,%d",
+            step_mem->stage_max_limit);
+
+    fprintf(outfile, ",Max. spectral radius,%.2f", step_mem->spr_max);
+    fprintf(outfile, ",Min. spectral radius,%.2f", step_mem->spr_min);
+    fprintf(outfile, "\n");
+    break;
+  default:
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Invalid formatting option.");
+    return (ARK_ILL_INPUT);
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  lsrkSSPStep_PrintAllStats:
+
+  Prints integrator statistics for SSP methods
+  ---------------------------------------------------------------*/
+int lsrkSSPStep_PrintAllStats(ARKodeMem ark_mem, FILE* outfile,
+                              SUNOutputFormat fmt)
+{
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeLSRKStepMem structure */
+  retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  switch (fmt)
+  {
+  case SUN_OUTPUTFORMAT_TABLE:
+    fprintf(outfile, "RHS fn evals                 = %ld\n", step_mem->nfe);
+    fprintf(outfile, "Current stages taken         = %d\n", step_mem->req_stages);
+    break;
+  case SUN_OUTPUTFORMAT_CSV:
+    fprintf(outfile, ",RHS fn evals,%ld", step_mem->nfe);
+    fprintf(outfile, ",Current stages taken,%d", step_mem->req_stages);
+
+    fprintf(outfile, "\n");
+    break;
+  default:
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Invalid formatting option.");
+    return (ARK_ILL_INPUT);
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_WriteParameters:
+
+  Outputs all solver parameters to the provided file pointer.
```

Comment:
Should this include other parameters like the eignevalue safety factor, max number of stages, etc.?

Reply:
Agreed -- this should output absolutely every internal parameter that the user can adjust.

Reply:
Do you mean that both `lsrkStep_PrintAllStats` and `lsrkSSPStep_PrintAllStats` should output all the parameters? I was following the same structure as with `ERKStep`, and it doesn't output the others either.

Reply:
To clarify this comment is for `WriteParameters` and it should print all the internal parameters for the method. `ERKStep` has very few parameters which is why its function is so short compared to the corresponding functions for `ARKStep` or `MRIStep`.

Reply:
It seems there there is a misunderstanding here between an input parameter and a stored statistic.  The `PrintAllStats` functions should print all of the statistics that are available from the stepper, but this `WriteParameters` routine should only output the current "control parameters" that a user could modify via "Set" routines.  It looks to me like the current version of this function includes multiple "output" statistics (e.g., `stage_max`, `step_nst`, `lambdaR`, `lambdaI`, ...).  All of those output statistics should *not* be printed from this routine.

To clarify the purpose of these different routines:
* `WriteParameters` is typically used at the **start** of a simulation, just before `ARKodeEvolve` is first called, so that the `stdout` file contains a complete snapshot of how the stepper was configured.
* `PrintAllStats` is typically used at the **end** of a simulation, and helps inform users of what occurred within the stepper as it solved the problem.